src/Microsoft.AspNet.Http/DefaultWebSocketManager.cs

```diff
@@ -27,12 +27,29 @@ void IFeatureCache.CheckFeaturesRevision()
         {
             if (_cachedFeaturesRevision != _features.Revision)
             {
-                _request = null;
-                _webSockets = null;
-                _cachedFeaturesRevision = _features.Revision;
+                ResetFeatures();
             }
         }
 
+        void IFeatureCache.SetFeaturesRevision()
+        {
+            _cachedFeaturesRevision = _features.Revision;
```

Comment:
Would it make sense to also set _cachedFeaturesRevision in the constructor? It would save you work on your first call to CheckFeaturesRevision.